### PR TITLE
Gh 185 spend transaction

### PR DESCRIPTION
**Pull request by zp-sd**
What is left:
* sign, put into the mempool and propagate to other peers transaction on `POST /spend-tx`
* change serialization to use msgpack, as it is done in https://github.com/aeternity/epoch/pull/310
* add sum of transaction fees to the miner's account when updating accounts state tree when creating/validating block candidate

### CODE REVIEW

File: apps/aeutils/src/aeu_validation.erl [hunk 1]
```diff
@@ -5,7 +5,7 @@
 run([], _Arg) ->
     ok;
 run([F | Rest], Arg) ->
-    case F(Arg) of
+    case apply(F, Arg) of
```
Comment (happi) [inline apps/aeutils/src/aeu_validation.erl]: To be a bit more explicit here Arg should be ArgList or something as I guess there are more than one argument since you need apply. (And the validation functions seem to take three arguments. Perhaps even be even more explicit and expect Arg to be a three element list and use F(Arg1, Arg2, Arg3) or even named arguments.

File: apps/aecore/src/txs/aec_spend_tx.erl [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    Trees = aec_trees:set_accounts(Trees0, AccountsTrees2),
+    {ok, Trees}.
+
+-spec serialize(spend_tx()) -> map().
+serialize(#spend_tx{sender = Sender,
+                    recipient = Recipient,
+                    amount = Amount,
+                    fee = Fee,
+                    nonce = Nonce}) ->
+    #{<<"sender">> => base64:encode(Sender),
```
Comment (uwiger) [inline apps/aecore/src/txs/aec_spend_tx.erl]: Please make this an array of maps instead, and without encoding:
```erlang
[#{<<"sender">> => Sender},
 #{<<"recipient">> => Recipient},
 ...]
```
C.f. https://github.com/aeternity/epoch/blob/gh-279-process-transactions/apps/aecore/src/txs/aec_coinbase_tx.erl#L54 (PR #310). Note also the inclusion of type and vsn.

File: apps/aehttp/priv/swagger.json [hunk 3]
```diff
@@ -148,6 +148,12 @@
         "responses" : {
           "200" : {
             "description" : "successful operation"
+          },
+          "400" : {
```
Comment (uwiger) [inline apps/aehttp/priv/swagger.json]: Elswhere in this spec, the code `404` has been used for error returns.

File: apps/aehttp/priv/swagger.json [hunk 4]
```diff
@@ -183,6 +189,36 @@
...
+          "required" : true,
+          "schema" : {
+            "$ref" : "#/definitions/SpendTx"
+          }
+        } ],
+        "responses" : {
+          "200" : {
+            "description" : "successful operation"
+          },
+          "400" : {
```
Comment (uwiger) [inline apps/aehttp/priv/swagger.json]: Elswhere in this spec, the code 404 has been used for error returns.

File: apps/aehttp/src/aehttp_dispatch_ext.erl [hunk 5]
```diff
@@ -134,6 +134,25 @@ handle_request('GetAccountBalance', Req, _Context) ->
...
+    Trees = aec_blocks:trees(LastBlock),
+    case aec_spend_tx:new(#{sender => SenderPubkey,
+                            recipient => RecipientPubkey,
+                            amount => Amount,
+                            fee => Fee}, Trees) of
+        {ok, _SpendTx} ->
+            %% TODO: sign and push to the mempool
+            {200, [], #{}};
+        {error, sender_account_not_found} ->
+            {400, [], #{reason => <<"sender account not found">>}}
```
Comment (uwiger) [inline apps/aehttp/src/aehttp_dispatch_ext.erl]: Should be `404` for consistency.

File: rebar.config [hunk 6]
```diff
@@ -24,7 +24,7 @@
 }.
 
 {plugins,
-  [{tx_xform, {git, "https://github.com/aeternity/tx_xform.git", {branch, "master"}}}]}.
+  [{tx_xform, {git, "https://github.com/zp-sd/tx_xform", {branch, "accept-error-response-from-aec_tx-new/2"}}}]}.
```
Comment (uwiger) [inline rebar.config]: The PR to `tx_xform` has been merged now, so pls revert this change.
